x-pack/plugin/migrate/src/main/java/org/elasticsearch/xpack/migrate/action/ReindexDataStreamAction.java

```diff
@@ -70,22 +81,55 @@ public boolean equals(Object other) {
 
     }
 
-    public static class ReindexDataStreamRequest extends ActionRequest {
+    public static class ReindexDataStreamRequest extends ActionRequest implements IndicesRequest, ToXContent {
+        private final String mode;
         private final String sourceDataStream;
 
-        public ReindexDataStreamRequest(String sourceDataStream) {
-            super();
+        public ReindexDataStreamRequest(String mode, String sourceDataStream) {
+            this.mode = mode;
             this.sourceDataStream = sourceDataStream;
         }
 
         public ReindexDataStreamRequest(StreamInput in) throws IOException {
             super(in);
+            this.mode = in.readString();
             this.sourceDataStream = in.readString();
         }
 
+        private static final ConstructingObjectParser<ReindexDataStreamRequest, Predicate<NodeFeature>> PARSER =
+            new ConstructingObjectParser<>("migration_reindex", new Function<Object[], ReindexDataStreamRequest>() {
+                @Override
+                public ReindexDataStreamRequest apply(Object[] objects) {
+                    String mode = (String) objects[0];
+                    String source = (String) objects[1];
+                    return new ReindexDataStreamRequest(mode, source);
+                }
+            });
+
+        private static final ConstructingObjectParser<String, Void> SOURCE_PARSER = new ConstructingObjectParser<>(
+            "source",
```

Comment:
We should pull this (and the `ParseField("index")` and `"mode"` below) into public static variables so that they are accessible from other classes. It's minor, but it will make it easier to refactor if we decide to change the fields later.

Reply:
Thanks for pulling it into fields, this one can change to 
```suggestion
            SOURCE_FIELD.getPreferredName(),
```
I think

Reply:
I actually think that's just a label for the parser and it doesn't matter what it is. But it can't hurt to set it to SOURCE_FIELD.getPreferredName(), so I'll do that.